src/components/activity-header/nav-pages.tsx

```diff
@@ -102,6 +124,9 @@ export class NavPages extends React.PureComponent <IProps> {
   }
 
   private handleChangePage = (page: number) => () => {
-    this.props.onPageChange(page);
+    if (!this.state.pageChangeInProgress) {
+      this.props.onPageChange(page);
+      this.setState({ pageChangeInProgress: true });
```

Comment:
Do you see any issue with setting the state after you call the callback versus calling it in the `setState` callback like:

`this.setState({ pageChangeInProgress: true }, () => this.props.onPageChange(page))`;

Reply:
To be super safe that callback can't be executed twice? 
I think it won't hurt. Some micro delay maybe, but I don't think it'd be noticeable by humans. ;) 
I made this change in the last comment, thanks.